waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -67,16 +63,42 @@ genJobs spec = return $ genJob <$> getJobs spec
                   [ "jobName" .= jobName,
                     "jobPerformFnName" .= jobPerformFnName,
                     "jobPerformFnImportStatement" .= jobPerformFnImportStatement,
-                    "jobFactoryName" .= show (jobFactoryForJob job)
+                    "jobExecutorFilename" .= toFilePath (basename $ jobCreatorFilePath $ executor job),
```

Comment:
this `basename` part here is a bit smelly to me -> We are assuming that these files (job creator files) are next to the job files.
What if they move? This will not recognize it.

Maybe we could use more of StrongPath around this? So you know, we define rel path from executor files to job dir, and then  use that to construct other paths, all while using StrongPath, so it is clear in type system / code that we are assuming directory structure to be in certain way?

Maybe I am overdoing this, not sure, but usually I try to push in this way with StrongPath. Sometimes it turns out to be too much, but often it really nicely "documents" assumptions.

Reply:
Yeah definitely, those are good points, I can give it a try with the other SP one 👍🏻